lib/error-handler.js

```diff
@@ -0,0 +1,151 @@
+'use strict'
+
+const FJS = require('fast-json-stringify')
+const statusCodes = require('http').STATUS_CODES
+const setErrorHeaders = require('./setErrorHeaders')
+const wrapThenable = require('./wrapThenable')
+const {
+  kReplyHeaders, kReplyNextErrorHandler, kReplyIsRunningOnErrorHook, kReplySent, kReplyHasStatusCode
+} = require('./symbols.js')
+
+const { getSchemaSerializer } = require('./schemas')
+
+const serializeError = FJS({
+  type: 'object',
+  properties: {
+    statusCode: { type: 'number' },
+    code: { type: 'string' },
+    error: { type: 'string' },
+    message: { type: 'string' }
+  }
+})
+
+const rootErrorHandler = {
+  func: defaultErrorHandler,
+  toJSON () {
+    return this.func.name.toString() + '()'
+  }
+}
+
+function handleError (reply, error, cb) {
+  console.log('handleError')
+  reply[kReplyIsRunningOnErrorHook] = false
+
+  const context = reply.context
+  if (reply[kReplyNextErrorHandler] === false) {
+    console.log('aaa')
+    fallbackErrorHandler(error, reply, function (reply, payload) {
+      reply.raw.writeHead(reply.raw.statusCode, reply[kReplyHeaders])
+      reply.raw.end(payload)
+    })
+    return
+  }
+  const errorHandler = reply[kReplyNextErrorHandler] || context.errorHandler
+
+  // In case we the error handler throws, we set the next errorHandler so we can error again
+  reply[kReplyNextErrorHandler] = Object.getPrototypeOf(errorHandler)
+
+  reply[kReplyHeaders]['content-length'] = undefined
+
+  const func = errorHandler.func
+  console.log(func)
```

Comment:
is this intentional?

Reply:
Yes. I'll remove all console.logs when I move it out of draft. Debugging this has been kind of hard and those statements helped me navigate.